### PR TITLE
Fix error on producing a block with multiple transaction on the same …

### PR DESCRIPTION
**Pull request by asdacap**
- Fix error on producing a block with multiple transaction for the same contract with block improvement.

## Changes

- Added unit test

#### What types of changes does your code introduce?

- [X] Bugfix (a non-breaking change that fixes an issue)
#### Requires testing

- [X] Yes
- [ ] No

#### If yes, did you write tests?

- [X] Yes
- [ ] No

#### Notes on testing

- Unit test can reproduce it.

### CODE REVIEW

File: src/Nethermind/Nethermind.Consensus/Processing/OneTimeProcessor.cs [hunk 1]
```diff
@@ -39,17 +39,7 @@ public Task StopAsync(bool processRemainingBlocks = false)
...
             lock (_lock)
             {
-                Block result;
-                try
-                {
-                    result = _processor.Process(block, options, tracer);
-                }
-                finally
-                {
-                    _worldState.Reset();
```
Comment (asdacap) [inline src/Nethermind/Nethermind.Consensus/Processing/OneTimeProcessor.cs:49]: Previously this was a `ReadOnlyDbProvider.ClearTempChanges()`. It does not actually do anything as the `AsReadOnly` have false as its parameter except for code db. Meaning only code db have buffer.

Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Consensus/Processing/OneTimeProcessor.cs:49]: Can it break something in RPC? like eth_call or tracing?

File: src/Nethermind/Nethermind.State/StateProvider.cs [hunk 2]
```diff
@@ -808,11 +808,6 @@ public void Reset()
             _currentPosition = Resettable.EmptyPosition;
             Array.Clear(_changes, 0, _changes.Length);
             _needsStateRootUpdate = false;
-
-            if (_codeDb is IReadOnlyDb readOnlyDb)
-            {
-                readOnlyDb.ClearTempChanges();
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.State/StateProvider.cs:814]: are those changes ever cleared? Not a leak?

Comment (asdacap) [reply src/Nethermind/Nethermind.State/StateProvider.cs:814]: Heck. I don't know.

File: src/Nethermind/Nethermind.Consensus/Processing/OneTimeProcessor.cs (hunk 1 above)
Comment (asdacap) [reply src/Nethermind/Nethermind.Consensus/Processing/OneTimeProcessor.cs:49]: Not sure, so I'm gonna be conservative and revert back to what it was.

File: src/Nethermind/Nethermind.Consensus/Processing/OneTimeProcessor.cs [hunk 3]
```diff
@@ -39,17 +39,7 @@ public Task StopAsync(bool processRemainingBlocks = false)
...
-                }
-
-                return result;
+                return _processor.Process(block, options, tracer);
```
Comment (MarekM25) [inline src/Nethermind/Nethermind.Consensus/Processing/OneTimeProcessor.cs:42]: hmmm but on version 1.24 we used to call `_readOnlyDbProvider.ClearTempChanges();`
https://github.com/NethermindEth/nethermind/blob/release/1.24.0/src/Nethermind/Nethermind.Consensus/Processing/OneTimeProcessor.cs

**Review by LukaszRozmej — APPROVED** [review]

**Review by kamilchodola — APPROVED** [review]
Works well on validator node

**Review by MarekM25 — APPROVED** [review]

**Pull request merged**